tests/fft.cpp

```diff
@@ -29,6 +29,31 @@ struct DFDim : ddc::PeriodicSampling<Kx>
 {
 };
 
+template <typename X, std::size_t Nx>
+static void test_fourier_mesh()
+{
+    double const a = -10;
+    double const b = 10;
+
+    DDom<DDim<X>> const x_mesh(ddc::init_discrete_space<DDim<X>>(DDim<X>::template init<DDim<X>>(
+            ddc::Coordinate<X>(a + (b - a) / Nx / 2),
+            ddc::Coordinate<X>(b - (b - a) / Nx / 2),
+            DVect<DDim<X>>(Nx))));
+    ddc::init_discrete_space<DFDim<ddc::Fourier<X>>>(
+            ddc::init_fourier_space<DFDim<ddc::Fourier<X>>>(ddc::DiscreteDomain<DDim<X>>(x_mesh)));
+
+    double const epsilon = 1e-14;
+    for (int i = 0; i < Nx; i++) {
+        double const ka = -Kokkos::numbers::pi * Nx / (b - a);
+        double const kb = Kokkos::numbers::pi * Nx / (b - a);
+        double const k = 2 * i * Kokkos::numbers::pi / (b - a);
+        EXPECT_NEAR(
+                ddc::coordinate(ddc::DiscreteElement<DFDim<ddc::Fourier<X>>>(i)),
+                k - (kb - ka) * Kokkos::floor((k - ka) / (kb - ka)),
```

Comment:
I do not get why we have to add this extra computation ?

Reply:
if `i>(N- N%2) / 2`, `ddc::coordinate` projects in the periodic window.

Initially I had the following test in mind (test only the points in the periodic window):

https://github.com/CExA-project/ddc/pull/541/commits/43124194762b83e26fd3603b1f57e43fb098ab9a#diff-3204c222afc48502013a0d15cc15f3e45be803127fb2f6f9720397ad7157e3bdL218

But it cannot work because `ddc` does not support `ddc::DiscreteElement(negative_integer)` (I mean, there is a cast involved. Actually the test passes, but by chance ^^. Using `std::ptrdiff_t` for `uid` types would fix the problem, but I like this new version of the test which indirectly also tests `PeriodicSampling`.